Assets/Prefabs/Player/PlayerController.cs

```diff
@@ -0,0 +1,173 @@
+using UnityEditor;
+using UnityEngine;
+
+public class PlayerController : MonoBehaviour
+{
+
+    #region UNITY_LIFECYCLE
+    // called when script instance is first loaded
+    private void Awake()
+    {
+        OnAwakeGetCharacterController();
+        OnAwakeBindInputs();
+    }
+
+    // called when GameObject is enabled
+    private void OnEnable()
+    {
+        inputActions.gameplay.Enable();
+    }
+
+    // called when GameObject is disabled
+    private void OnDisable()
+    {
+        inputActions.gameplay.Disable();
+    }
+    
+    // called on every Physics update
+    private void FixedUpdate()
+    {
+        ApplyMotion();
+    }
+    #endregion
+
+    #region PLAYER_INPUT
+    private InputActions inputActions;
+    private Vector2 moveInput = Vector2.zero;
+    private float moveAngle = 0.0f;
+    private float moveMagnitude = 0.0f;
+
+    /// <summary>
+    /// Instantiate the <c>InputActions</c> class and bind callbacks to store the 
+    /// current state of player input on changes.
+    /// </summary>
+    /// /// <remarks>Must be called during <c>OnAwake()</c></remarks>
+    private void OnAwakeBindInputs()
+    {
+        inputActions = new InputActions();
+        inputActions.gameplay.move.performed += context =>
+        {
+            moveInput = context.ReadValue<Vector2>();
+            moveAngle = -Vector2.SignedAngle(Vector2.up, moveInput);
+            moveMagnitude = moveInput.magnitude;
+        };
+        inputActions.gameplay.move.canceled += context =>
+        {
+            moveInput = Vector2.zero;
+            moveAngle = 0.0f;
+            moveMagnitude = 0.0f;
+        };
+    }
+
+    #endregion
+
+    #region PLAYER_MOVEMENT
+    [Header("Player Movement")]
+
+    [Tooltip("Camera to perform player motion relative to")]
+    public Camera myCamera;
+    [Tooltip("Walking speed of character (units / sec)")]
+    public float walkSpeed = 5.0f;
+    [Tooltip("Amount of time it takes to reach full walk / run speed (sec)")]
+    public float speedUpTime = 0.2f;
+    [Tooltip("Time it takes for the character to stop when the joystick is released (sec)")]
+    public float stopTime = 0.1f;
+    // TODO: IMPLEMENT THIS IN TERMS OF TIME BY SCALING WITH TIME.DELTATIME
+    [Tooltip("Portion of the rotatin between curent and desired orientation to apply each physics update"), Range(0, 1.0f)]
+    public float rotationRatio = 0.2f;
+    [Tooltip("Gravitational Constant (units / sec / sec)")]
+    public float gravity = -10.0f;
```

Comment:
For the current gameplay, a gravity of -0.1 is more fun / has a smoother mechanic. Though, of course this is something that will be modified as we go depending on the style.

Reply:
This will definitely need to be adjusted as we build up levels - but I agree a much weaker gravity feels more fun, so I changed it to `0.3` for now!